code/game/objects/items/weapons/dice.dm

```diff
@@ -122,3 +122,169 @@
 				for(var/obj/machinery/computer/bhangmeter/bhangmeter in doppler_arrays)
 					if(bhangmeter)
 						bhangmeter.sense_explosion(epicenter.x,epicenter.y,epicenter.z,round(uncapped*0.25), round(uncapped*0.5), round(uncapped),"???", cap)
+
+
+
+/obj/item/weapon/dice/d20/cursed
+	desc = "Something about this dice seems wrong"
+	name = "Mysterious d20"
+	var/deactivated = 0 //Eventually the dice runs out of power
+
+/obj/item/weapon/dice/d20/cursed/pickup(mob/user as mob)
+	if(deactivated == 0)
+		to_chat(user, "<span class=sinister><B>Are you feeling lucky?</span></B>")
+	else
+		return 0
+
+/obj/item/weapon/dice/d20/cursed/diceroll(mob/user as mob, thrown)
+	..()
+	if(deactivated == 0) //If the dice has power then something will happen
+		message_admins("[key_name(user)] has [thrown? "used" : "thrown"] a cursed dice and rolled a [result]")
+		log_game("[key_name(user)] has [thrown? "used" : "thrown"] a cursed dice and rolled a [result]")
+		if(istype(user,/mob/living/carbon/human))
+			var/mob/living/carbon/human/h = user
+			if(result == 1)

```

Comment:
Make this gigantic tree of if statements into switch statements.

With a switch, you can do `if(x to y)`.